runtime-modules/content/src/lib.rs

```diff
@@ -2256,6 +2100,59 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    pub fn ensure_valid_init_transactional_status(
+        init_status: &InitTransactionalStatus<T>,
+    ) -> Result<
+        TransactionalStatus<
+            <T as frame_system::Trait>::BlockNumber,
+            <T as common::MembershipTypes>::MemberId,
+            <T as frame_system::Trait>::AccountId,
+            CurrencyOf<T>,
+        >,
+        DispatchError,
+    > {
+        match init_status {
+            InitTransactionalStatus::<T>::Idle => Ok(TransactionalStatus::Idle),
+            InitTransactionalStatus::<T>::InitiatedOfferToMember(member, balance) => Ok(
+                TransactionalStatus::InitiatedOfferToMember(member.clone(), balance.clone()),
+            ),
+            InitTransactionalStatus::<T>::Auction(params) => {
+                Self::validate_auction_params(&params)?;
+                let mut auction = AuctionRecord::new(params.clone());
+                if params.starts_at.is_none() {
+                    auction.starts_at = <frame_system::Module<T>>::block_number();
+                }
+                Ok(TransactionalStatus::Auction(auction))
+            }
+        }
+    }
+
+    pub fn construct_owned_nft(
```

Comment:
Do we need this method as public?

Reply:
fixed in 92a405cf94a779685a5ed4f648210345e00a75fa